src/kOS/Communication/InterVesselManager.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using kOS.Safe.Encapsulation;
+using kOS.Safe.Serialization;
+using kOS.Serialization;
+using kOS.Safe;
+
+namespace kOS.Communication
+{
+    [KSPScenario(ScenarioCreationOptions.AddToAllGames | ScenarioCreationOptions.AddToExistingGames, GameScenes.FLIGHT, GameScenes.SPACECENTER, GameScenes.TRACKSTATION)]
+    public class InterVesselManager : ScenarioModule
+    {
+        private const string Id = "id";
+        private static string VesselQueue = "vesselQueue";
+        private static string MessageQueue = "messageQueue";
+
+        private Dictionary<string, MessageQueue> vesselQueues;
+
+        public static InterVesselManager Instance { get; private set; }
+
+        static InterVesselManager() {
+            // normally we do this in SerializationMgr, but KSPScenarios run before we create any instances
+            SafeSerializationMgr.AddAssembly(typeof(SerializationMgr).Assembly.FullName);
+        }
```

Comment:
(See elsewhere in the PR comments for the comment with the word "curly brace" in big letters.)